### PR TITLE
[FIX] App forgetting workspace when server is not finished added

### PR DESCRIPTION
**Pull request by gerzonc**
## Proposed changes

Modifies the way app manages sessions in order to keep them persistent by adding a validation where it checks if there are any other logged in servers within the app. If so it will pick the first one (which is the last one you logged in).

## Issue(s)	

Closes #2122, which includes steps on how to reproduce this issue.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/sagas/init.js [hunk 1]
```diff
@@ -25,16 +28,30 @@ export const initLocalSettings = function* initLocalSettings() {
 
 const restore = function* restore() {
 	try {
-		const { token, server } = yield all({
-			token: UserPreferences.getStringAsync(RocketChat.TOKEN_KEY),
-			server: UserPreferences.getStringAsync(RocketChat.CURRENT_SERVER)
-		});
+		const server = yield UserPreferences.getStringAsync('currentServer');
+		let userId = server
```
Comment (diegolmello) [inline app/sagas/init.js:32]: You check for the existence of `server` here and applies `null` to `userId`.
After that you check for `server` again and dispatch `appStart`.
Can we check for the server only once?
It'd simplify this logic.

File: app/sagas/login.js [hunk 2]
```diff
@@ -126,12 +126,12 @@ const fetchRooms = function* fetchRooms() {
...
 	try {
 		const adding = yield select(state => state.server.adding);
-		yield UserPreferences.setStringAsync(RocketChat.TOKEN_KEY, user.token);
+
 
 		RocketChat.getUserPresence(user.id);
 
 		const server = yield select(getServer);
-		yield fork(fetchRooms);
+		yield fork(fetchRooms, { server });
```
Comment (diegolmello) [inline app/sagas/login.js:134]: What are you doing here?

File: app/sagas/init.js [hunk 3]
```diff
@@ -25,16 +28,30 @@ export const initLocalSettings = function* initLocalSettings() {
 
 const restore = function* restore() {
 	try {
-		const { token, server } = yield all({
-			token: UserPreferences.getStringAsync(RocketChat.TOKEN_KEY),
-			server: UserPreferences.getStringAsync(RocketChat.CURRENT_SERVER)
-		});
+		const server = yield UserPreferences.getStringAsync('currentServer');
```
Comment (diegolmello) [inline app/sagas/init.js:31]: Why did you change from `UserPreferences.getStringAsync(RocketChat.CURRENT_SERVER)` to the string version?

File: app/sagas/login.js [hunk 4]
```diff
@@ -126,12 +126,12 @@ const fetchRooms = function* fetchRooms() {
 const handleLoginSuccess = function* handleLoginSuccess({ user }) {
 	try {
 		const adding = yield select(state => state.server.adding);
-		yield UserPreferences.setStringAsync(RocketChat.TOKEN_KEY, user.token);
+
```
Comment (diegolmello) [inline app/sagas/login.js:129]: Are you sure about this?

File: app/views/RoomsListView/Header/Header.js [hunk 5]
```diff
@@ -82,7 +82,7 @@ const Header = React.memo(({
 						size={18}
 					/>
 				</View>
-				{subtitle ? <Text style={[styles.subtitle, { color: themes[theme].auxiliaryText, fontSize: subTitleFontSize }]} numberOfLines={1}>{subtitle}</Text> : null}
+				{subtitle ? <Text testID='rooms-list-header-server-subtitle' style={[styles.subtitle, { color: themes[theme].auxiliaryText, fontSize: subTitleFontSize }]} numberOfLines={1}>{subtitle}</Text> : null}
```
Comment (diegolmello) [inline app/views/RoomsListView/Header/Header.js:85]: You added testID here, but you didn't mention it anywhere else in your changes.
Is this correct?

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/sagas/init.js (hunk 1 above)
Comment (gerzonc) [reply app/sagas/init.js]: Done.

File: app/sagas/login.js (hunk 2 above)
Comment (gerzonc) [reply app/sagas/login.js]: This is wrong, reverted.

File: app/sagas/login.js (hunk 4 above)
Comment (gerzonc) [reply app/sagas/login.js]: Reverted

File: app/views/RoomsListView/Header/Header.js (hunk 5 above)
Comment (gerzonc) [reply app/views/RoomsListView/Header/Header.js:85]: This is correct, it was added for the e2e tests.

**Review by diegolmello — APPROVED** [review]
@gerzonc Removed a couple of calls to RocketChat.TOKEN_KEY and changed the tests a little. Check it out :)
